front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/PipelineController.groovy

```diff
@@ -124,21 +133,68 @@ class PipelineController {
     if (pipeline.id != existingPipeline.id) {
       throw new InvalidRequestException("The provided id ${id} doesn't match the pipeline id ${pipeline.id}")
     }
-    pipeline.name = pipeline.getName().trim()
 
-    if (pipelineDAO.getPipelinesByApplication(pipeline.getApplication()).any {
-      it.getName().equalsIgnoreCase(pipeline.getName()) && it.getId() != id }) {
-      throw new DuplicateEntityException("A pipeline with name ${pipeline.getName()} already exists in application ${pipeline.application}")
-    }
+    validatePipeline(pipeline)
```

Comment:
This is after checking that the pipeline ids match which is after a datastore hit I could move this validate to the first thing in the method but that changes error cases in the TCK.